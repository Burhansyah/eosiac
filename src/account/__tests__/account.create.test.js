const cloneDeep = require(`lodash/cloneDeep`)
const Account = require(`../index`)
const setup = require(`./setup`)

const {env, accountConfig, existingAccount} = setup

describe(`account.create`, () => {
    let newEnv
    beforeEach(() => {
        newEnv = cloneDeep(env)
        newEnv.accounts.accntmngr = new Account(`accntmngr`, newEnv.accounts.accntmngr)
        newEnv.accounts.accntmngr.currentState = {
            ram_quota: 100,
        }
    })
    it(`returns new account action on empty current state`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = {}
        const actions = await a.create({env: newEnv})
        expect(actions).toMatchSnapshot()
    })

    it(`returns no action on existing currentState`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        const actions = await a.create({env: newEnv})
        expect(actions).toBeFalsy()
    })
})
