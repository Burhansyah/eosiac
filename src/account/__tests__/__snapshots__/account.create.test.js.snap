// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account.create returns new account action on empty current state 1`] = `
Array [
  Object {
    "account": "eosio",
    "authorization": Array [
      Object {
        "actor": "accntmngr",
        "permission": "active",
      },
    ],
    "data": Object {
      "active": Object {
        "accounts": Array [
          Object {
            "permission": Object {
              "actor": "dapptoken",
              "permission": "eosio.code",
            },
            "weight": 1,
          },
        ],
        "keys": Array [
          Object {
            "key": "EOS5nYs7LDcFcVtFmnYrhYQv9Z8DPDd1r4MJ8na2MvcPPJkXwzM3x",
            "weight": 1,
          },
        ],
        "parent": "owner",
        "threshold": 2,
        "waits": Array [
          Object {
            "wait_sec": 600,
            "weight": 1,
          },
        ],
      },
      "creator": "accntmngr",
      "name": "test",
      "owner": Object {
        "accounts": Array [],
        "keys": Array [
          Object {
            "key": "EOS6z98haKv12d8vAJbDUkXmpK3PaKvf3Dv1NApo9MCxJ8oNWssDi",
            "weight": 1,
          },
        ],
        "parent": "",
        "threshold": 1,
        "waits": Array [],
      },
    },
    "name": "newaccount",
  },
  Object {
    "account": "eosio",
    "authorization": Array [
      Object {
        "actor": "rammngr",
        "permission": "active",
      },
    ],
    "data": Object {
      "bytes": 3072,
      "payer": "rammngr",
      "receiver": "test",
    },
    "name": "buyrambytes",
  },
]
`;
